lib/policy_machine.rb

```diff
@@ -127,7 +126,9 @@ def is_privilege_ignoring_prohibitions?(user_or_attribute, operation, object_or_
       raise(ArgumentError, "options[:associations] cannot be empty") if associations.empty?
       raise(ArgumentError, "expected each element of options[:associations] to be a PM::Association") unless associations.all?{|a| a.is_a?(PM::Association)}
 
-      associations.keep_if{ |assoc| assoc.includes_operation?(operation) }
+      associations.keep_if do |association|
+        association.operation_set.connected?(operation)
```

Comment:
Later. It _looks_ to me like this is really only intended to be used in tests. We could use the `filter_operation_set_list_by_assigned_operation` method to remove the n+1 query when the storage adapter is psql, but that would require writing adapter-specific logic into this method. Again, that doesn't seem worth it to me right now (since we and no one in the foreseeable future) will ever use this, so I'm inclined to leave it as is unless objections.